docs/concepts/projects.md

```diff
@@ -203,6 +217,15 @@ Anytime you run `prefect deploy`, this `push` section will be executed upon succ
 
 The pull section is the most important section within the `prefect.yaml` file as it contains instructions for preparing this project for a deployment run.  These instructions will be executed each time a deployment created wthin this project is run via a worker.
 
+There are three main types of steps that typically show up in a `pull` section:
+
+- `set_working_directory`: this step simply sets the working directory for the process prior to importing your flow
+- `git_clone_project`: this step clones the provided repository on the provided branch
+- `pull_project_from_{cloud}`: this step pulls the project directory from a Cloud storage location (e.g., S3)
+
+!!! tip "Use block and variable references"
+    All [block and variable references](#templating-options) within your pull step will remain unresolved until runtime and will be pulled each time your deployment is run. This allows you to avoid storing sensitive information in an insecure way; it also allows you to manage certain types of configuration from the API and UI without having to rebuild your deployment every time.
```

Comment:
```suggestion
    All [block and variable references](#templating-options) within your pull step will remain unresolved until runtime and will be pulled each time your deployment is run. This allows you to avoid storing sensitive information insecurely; it also allows you to manage certain types of configuration from the API and UI without having to rebuild your deployment every time.
```